notebooks/ai-ml/SparkXGBoostCustomerChurn.ipynb

```diff
@@ -0,0 +1,838 @@
+    "\n",
+    "assembler = VectorAssembler(inputCols=[c + \"_encoded\" for c in categorical_cols] + numerical_cols, outputCol=\"features\", handleInvalid=\"keep\") \n",
+    "\n",
+    "# Apply transformations:\n",
+    "indexed_df = Pipeline(stages=indexers).fit(df_train).transform(df_train)\n",
+    "encoded_df = Pipeline(stages=encoders).fit(indexed_df).transform(indexed_df)\n",
+    "transformed_data = assembler.transform(encoded_df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dc49edcb-1b2f-40f6-b08d-31cc18be7db4",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Train the model and generate predictions\n",
+    "\n",
+    "Use the [SparkXGBoostClassifier](https://xgboost.readthedocs.io/en/latest/tutorials/spark_estimator.html#sparkxgbclassifier) to create a SparkML model. This is passed into a SparkML [pipeline](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.Pipeline.html) object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8bedfb96-f79f-4eb1-a163-b1fc719089de",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "xgb = SparkXGBClassifier(label_col=\"churn\", features_col=\"features\", num_workers=2, missing=0.0)\n",
+    "\n",
+    "pipeline = Pipeline(stages=[xgb])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88dae1ea-b329-4aba-bedc-aed7ff8a1e55",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Split the data into training and test datasets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d648b41-a6a8-4b67-9fce-ca5c4475c790",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "train_data, test_data = transformed_data.randomSplit([0.8, 0.2], seed=42) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c993d2a1-6ea2-471d-9289-fe4844e8a624",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "Fit the model using the training data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3ec8fe9a-f1f6-490f-acb3-6978621913cd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = pipeline.fit(train_data)\n",
+    "print(\"Model training successful.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d3d4818-d509-4089-961f-eb98377eefa8",
+   "metadata": {},
+   "source": [
+    "Generate predictions on the test data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "98238a24-3545-4fea-acc8-09d696c73d1e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictions = model.transform(test_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c094288c-8f29-463b-9634-e5738c6d2188",
+   "metadata": {},
+   "source": [
+    "## Evaluate the model.\n",
+    "\n",
+    "Evaluate the performance of the model. The metrics used are:\n",
```

Comment:
Give a 1 sentence description of what these metrics measure and why they matter. I like that you link to each one as well! Though if PRC doesn't have a link, it might be okay to omit too - we have quite a bit of eval metrics included.